From db8367028c994ead1b0c60cfbe3d03c8140181e3 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 20 May 2020 10:24:44 +0300
Subject: [PATCH] [SW-0] spi: do not (re)write the config register

During the Status Register write operation, we can also write the
Configuration Register, but we do not have to, as SR write operation can
end also after 8 bits of the SR content.
Remove redundant CR read and further CR write of the same content.

Change-Id: I2afe32fb723772bbb5d50382d12e46b8f9a26b1a
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/spi.c b/common/spi.c
index 6033594..8e579cd 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -31,7 +31,6 @@
 #define CMD_WRITE_DISABLE		0x04
 #define CMD_READ_STATUS			0x05
 #define CMD_WRITE_ENABLE		0x06
-#define CMD_READ_CONF			0x15
 #define READ_STATUS_WIP			BIT(0)
 #define READ_STATUS_BP0			BIT(2)
 #define READ_STATUS_BP1			BIT(3)
@@ -186,14 +185,12 @@
 
 static unsigned char spi_prot(void)
 {
-	unsigned char sr, cr, data[2];
+	unsigned char sr;
 
-	spi_char_cmd_csel(CMD_READ_CONF, &cr);
 	spi_char_cmd_csel(CMD_READ_STATUS, &sr);
 	sr |= PROT_MASK;
-	data[0] = sr;
-	data[1] = cr;
-	spi_common_write(CMD_WRITE_STATUS, data, sizeof(data));
+	spi_common_write(CMD_WRITE_STATUS, &sr, sizeof(sr));
+
 	return sr;
 }
 
